src/context/uuid.rs

```diff
@@ -0,0 +1,57 @@
+use crate::ContextUuidExt;
+
+use super::Context;
+
+/// Describes an error preventing the retrieval of the uuid.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum UuidError {
+    /// EXT_external_objects is not supported by the driver
+    ExtensionNotPresent,
+}
+
+impl std::fmt::Display for UuidError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let desc = match self {
+            UuidError::ExtensionNotPresent => "EXT_external_objects is not supported by the driver",
+        };
+
+	f.write_str(desc)
+    }
+}
+
+impl std::error::Error for UuidError {}
+
+
+
+impl ContextUuidExt for Context {
```

Comment:
This trait is only implemented once, right? Why is it an extra trait and why is it not separate functions on the `Context` struct?

Reply:
Yes, it's only implemented once and probably will never be implemented by any other struct. 

Since most glium users likely won't be using this functionality, I thought it would be a good idea to compartmentalize it and make it opt-in via the use of an extension trait. Of course the drawbacks are the added verbosity in the library as well as some user programs having to use an additional line `use glium::ContextUuidExt`.

If you think the drawbacks outweigh the benefits, let me know and I can make the methods part of the `Context` struct.

Reply:
I think it's better they become parts of the `Context` struct. `Context` already has a ton of functions, these two won't add too much noise. The impl block can stay in this file.